### PR TITLE
Member committeemeeting attendance

### PR DESCRIPTION
**Pull request by Lunga001**

### CODE REVIEW

**Comment by Lunga001** [comment]
Members - PMG-CM

**Comment by Lunga001** [comment]
Members-PMG-CMS

File: pmg/admin/__init__.py [hunk 1]
```diff
@@ -847,6 +847,62 @@ def on_model_change(self, form, model, is_created):
...
+        return redirect(url)
+
+
+class MemberCommitteeMeetingAttendanceView(MyModelView):
+    column_list = (
+        "meeting.committee.name",
+        "meeting.date",
+        "meeting.title",
+        "attendance",
+        "alternate_member",
```
Comment (jbothma) [inline pmg/admin/__init__.py:865]: add member name

File: pmg/admin/__init__.py [hunk 2]
```diff
@@ -847,6 +847,62 @@ def on_model_change(self, form, model, is_created):
...
+    )
+    column_labels = {
+        "meeting.date": "Meeting date",
+        "meeting.title": "Meeting title",
+        "meeting.committee.name": "Committee name",
+        "attendance": "Attendance code",
+    }
+    column_searchable_list = ("attendance",)
```
Comment (jbothma) [inline pmg/admin/__init__.py:873]: member name
committee name

Comment (jbothma) [reply pmg/admin/__init__.py:873]: if it takes more than 5 mins, just make it not show up at all (empoty?)

File: pmg/admin/__init__.py [hunk 3]
```diff
@@ -847,6 +847,62 @@ def on_model_change(self, form, model, is_created):
...
+    column_default_sort = ("meeting_id", True)
```
Comment (jbothma) [inline pmg/admin/__init__.py:874]: remove this

File: pmg/admin/__init__.py [hunk 4]
```diff
@@ -847,6 +847,62 @@ def on_model_change(self, form, model, is_created):
...
+    column_filters = ["meeting.date", "meeting.committee"]
+    column_formatters = {
+        "meeting.committee.name": lambda v, c, m, n: Markup(
+            "<a href='%s'>%s</a>"
+            % (
+                url_for("committee_meeting", event_id=m.meeting_id),
+                m.meeting.committee.name,
+            ),
+        ),
```
Comment (jbothma) [inline pmg/admin/__init__.py:883]: if it take more than 10 minutes to get this right in the template, leave it as is

File: pmg/admin/__init__.py [hunk 5]
```diff
@@ -847,6 +847,62 @@ def on_model_change(self, form, model, is_created):
...
+    def get_count_query(self):
+        return self._extend_query(
+            super(MemberCommitteeMeetingAttendanceView, self).get_count_query()
+        )
+
+    def _extend_query(self, query):
+        member_id = request.args.get("member_id")
+        if member_id is None:
+            abort(400, "Member required")
+        return query.filter(CommitteeMeetingAttendance.member_id == member_id)
```
Comment (jbothma) [inline pmg/admin/__init__.py:904]: if it doesn't take more than 10 minutes, show under committee meetings here. It should def not be in main menu.
Screenshot_2021-06-17_15-26-11

File: pmg/admin/__init__.py [hunk 6]
```diff
@@ -847,6 +847,62 @@ def on_model_change(self, form, model, is_created):
...
+            % (
+                url_for("committee_meeting", event_id=m.meeting_id),
+                m.meeting.committee.name,
+            ),
+        ),
+        "meeting.date": lambda v, c, m, n: m.meeting.date.date().isoformat(),
+    }
+
+    def is_visible(self):
+        return False
```
Comment (jbothma) [inline pmg/admin/__init__.py:888]: if it doesn't take more than 10 minutes, show under committee meetings here. It should def not be in main menu.
Screenshot_2021-06-17_15-26-11

File: pmg/admin/__init__.py [hunk 7]
```diff
@@ -847,6 +847,62 @@ def on_model_change(self, form, model, is_created):
...
+        return self._extend_query(
+            super(MemberCommitteeMeetingAttendanceView, self).get_count_query()
+        )
+
+    def _extend_query(self, query):
+        member_id = request.args.get("member_id")
+        if member_id is None:
+            abort(400, "Member required")
+        return query.filter(CommitteeMeetingAttendance.member_id == member_id)
+
```
Comment (jbothma) [inline pmg/admin/__init__.py:905]: if no id, just show all. don't filter

File: pmg/admin/__init__.py [hunk 8]
```diff
@@ -847,6 +847,62 @@ def on_model_change(self, form, model, is_created):
...
+    )
+    column_labels = {
+        "meeting.date": "Meeting date",
+        "meeting.title": "Meeting title",
+        "meeting.committee.name": "Committee name",
+        "attendance": "Attendance code",
+    }
+    column_searchable_list = ("attendance",)
+    column_default_sort = ("meeting_id", True)
+    column_filters = ["meeting.date", "meeting.committee"]
```
Comment (jbothma) [inline pmg/admin/__init__.py:875]: attendance columns should also be filterable

File: pmg/admin/__init__.py [hunk 9]
```diff
@@ -847,6 +847,60 @@ def on_model_change(self, form, model, is_created):
...
+        "member.name",
+    )
+    column_labels = {
+        "meeting.date": "Meeting date",
+        "meeting.title": "Meeting title",
+        "meeting.committee.name": "Committee name",
+        "attendance": "Attendance code",
+        "member.name": "Member",
+    }
+    column_searchable_list = ("attendance", "member.name")
```
Comment (jbothma) [inline pmg/admin/__init__.py:875]:
```suggestion
    column_searchable_list = ("member.name",)
```

**Review by jbothma — APPROVED** [review]

**Pull request merged**